optuna/study.py

```diff
@@ -71,8 +71,7 @@ def best_value(self):
         """
 
         best_value = self.best_trial.value
-        if best_value is None:
```

Comment:
This condition never holds because,
- `self.best_trial` always returns a complete trial, and
- a complete trial doesn't have a `None` value.